### PR TITLE
Add Bool type to IR

### LINKED ISSUE #8758: [JIT] Add bool type to the schema (fix torch.var)
`torch.var` and `torch.std` in jit script exhibit incorrect behavior.

```
import torch
@torch.jit.script
def fn(x):
    dim = 1
    return torch.std(x, dim)

x = torch.randn(2, 3)
fn(x)  # gives  tensor(0.8668)
torch.std(x, 1)  # gives tensor([1.2670, 0.4717])
```

This is because of the following:

there are two schemas for var (and std):
(1) var(Tensor, int, *, bool, bool, Tensor)
(2) var(Tensor, *, bool)

Right now, the following is happening:
- Shorter schemas come before longer schemas
- bool, int are treated as IntType rather than DynamicType like before
So the schemas look like the following in aten_schema:
(2) var(DynamicType, IntType)
(1) var(DynamicType, IntType, IntType, DynamicType)

Now, when one calls torch.var(tensor, dim=1), the compiler mistakingly matches it with (2) instead of (1), which is a problem.

### PR DESCRIPTION
**Pull request by driazati**
Fixes #8758

* Adds explicit bool type and removes previous coercion from bool to int
* A couple tests now pass as a result

### CODE REVIEW

**Review by apaszke — COMMENTED** [review]
I'm a bit worried that adding a new attribute kind will make export more complicated, since the ONNX protobuf doesn't have boolean attributes IIRC.

**Comment by driazati** [comment]
The motivation for this was that some of the overloaded `aten` operators are ambiguous without an explicit bool type:

```
aten::std(Tensor self, bool unbiased=True) -> Tensor
aten::std(Tensor self, int dim, bool unbiased=True, bool keepdim=False) -> Tensor
```
